application/src/main/java/org/thingsboard/server/controller/DashboardController.java

```diff
@@ -107,17 +108,61 @@ public Dashboard getDashboardById(@PathVariable(DASHBOARD_ID) String strDashboar
         }
     }
 
+    @PreAuthorize("hasAnyAuthority('TENANT_ADMIN', 'CUSTOMER_USER')")
+    @RequestMapping(value = "/dashboard/{dashboardId}/entityConfig/{entityConfigId}/restore", method = RequestMethod.POST)
+    @ResponseBody
+    public Dashboard restoreDashboardConfig(@PathVariable(DASHBOARD_ID) String strDashboardId,
+                                            @PathVariable("entityConfigId") String strEntityConfigId,
+                                            @RequestParam(required = false) String comment) throws ThingsboardException {
+        checkParameter(DASHBOARD_ID, strDashboardId);
+        checkParameter("entityConfigId", strEntityConfigId);
+        try {
+            DashboardId dashboardId = new DashboardId(toUUID(strDashboardId));
+            checkDashboardId(dashboardId, Operation.READ);
+            ObjectNode additionalInfo = getEntityConfigAdditionalInfo(comment);
+            return dashboardService.restoreDashboardConfig(getTenantId(), dashboardId, new EntityConfigId(toUUID(strEntityConfigId)), additionalInfo);
+        } catch (Exception e) {
+            throw handleException(e);
+        }
+    }
+
+    @PreAuthorize("hasAnyAuthority('TENANT_ADMIN', 'CUSTOMER_USER')")
+    @RequestMapping(value = "/dashboard/{dashboardId}/entityConfig/{entityConfigId}/restoreAsNew", method = RequestMethod.POST)
+    @ResponseBody
+    public Dashboard restoreDashboardAsNew(@PathVariable(DASHBOARD_ID) String strDashboardId,
+                                            @PathVariable("entityConfigId") String strEntityConfigId,
+                                            @RequestParam(required = false) String comment,
+                                            @RequestParam(required = false) String title) throws ThingsboardException {
+        checkParameter(DASHBOARD_ID, strDashboardId);
+        checkParameter("entityConfigId", strEntityConfigId);
+        try {
+            DashboardId dashboardId = new DashboardId(toUUID(strDashboardId));
+            Dashboard dashboard = checkDashboardId(dashboardId, Operation.WRITE);
+            checkDashboardId(dashboardId, Operation.READ);
```

Comment:
you check for write permissions on the previous line. Why do you need another check for read?